frontend/src/metabase/forms/components/Form/Form.tsx

```diff
@@ -9,13 +11,22 @@ export interface FormProps
   extends BoxProps,
     FormHTMLAttributes<HTMLFormElement> {
   disabled?: boolean;
+  onValuesChange?: (vals: any) => void;
 }
 
 export const Form = forwardRef(function Form(
-  { disabled, ...props }: FormProps,
+  { disabled, onValuesChange, ...props }: FormProps,
   ref: Ref<HTMLFormElement>,
 ) {
-  const { handleSubmit, handleReset } = useFormikContext();
+  const { handleSubmit, handleReset, values } = useFormikContext();
+
+  const previousValues = usePrevious(values);
+
+  useEffect(() => {
+    if (!_.isEqual(previousValues, values)) {
+      onValuesChange?.(values);
+    }
+  }, [previousValues, values, onValuesChange]);
```

Comment:
can we avoid this? it looks like we need it to spy on the state of the shallow copy checkbox to modify the modal title. i fear the addition of this useEffect and usePrevious hook is adding performance overhead to all our forms for the sake of this one form.  we should be able to use the `useField` hook to get the changed state of that field in this one form, right?  maybe an argument for making this whole thing it's own modal now rather than later.

Reply:
There are fair points. I pulled this logic from https://github.com/metabase/metabase/blob/master/frontend/src/metabase/containers/FormikForm/FormikFormViewAdapter.tsx#L67, but it can likely be improved. 

I don't believe you can use `useField` outside of the form provider, so that likely won't work for us here. This is also going to come up with https://github.com/metabase/metabase/issues/41539, where we need to react to when a specific form prop changes as well.

The general consensus from the Formik library seems to be to create some kind of observer component that will largely handle this, and add that to your form (though people are cranky about it haha). This approach wouldn't look *super* different from what's here except it would allow us to only add this in places where we care about it, narrowing the scope of change.